base/strings/annotated.jl

```diff
@@ -386,3 +386,94 @@ annotations(s::SubString{<:AnnotatedString}, pos::UnitRange{<:Integer}) =
 Get all annotations of `chr`.
 """
 annotations(c::AnnotatedChar) = c.annotations
+
+## AnnotatedIOBuffer
+
+struct AnnotatedIOBuffer <: IO
+    io::IOBuffer
+    annotations::Vector{Tuple{UnitRange{Int}, Pair{Symbol, Any}}}
+end
+
+AnnotatedIOBuffer(io::IOBuffer) = AnnotatedIOBuffer(io, Vector{Tuple{UnitRange{Int}, Pair{Symbol, Any}}}())
+AnnotatedIOBuffer() = AnnotatedIOBuffer(IOBuffer())
+
+function show(io::IO, aio::AnnotatedIOBuffer)
+    show(io, AnnotatedIOBuffer)
+    print(io, '(', aio.io.size, " byte", ifelse(aio.io.size == 1, "", "s"), ", ",
+          length(aio.annotations), " annotation", ifelse(length(aio.annotations) == 1, "", "s"), ")")
+end
+
+isopen(io::AnnotatedIOBuffer) = isopen(io.io)
+close(io::AnnotatedIOBuffer) = close(io.io)
+closewrite(io::AnnotatedIOBuffer) = closewrite(io.io)
+eof(io::AnnotatedIOBuffer) = eof(io.io)
+peek(io::AnnotatedIOBuffer) = peek(io.io)
+position(io::AnnotatedIOBuffer) = position(io.io)
+seek(io::AnnotatedIOBuffer, n::Integer) = (seek(io.io, n); io)
+skip(io::AnnotatedIOBuffer, n::Integer) = (skip(io.io, n); io)
+mark(io::AnnotatedIOBuffer) = mark(io.io)
+reset(io::AnnotatedIOBuffer) = reset(io.io)
+unmark(io::AnnotatedIOBuffer) = unmark(io.io)
+ismarked(io::AnnotatedIOBuffer) = ismarked(io.io)
+copy(io::AnnotatedIOBuffer) = AnnotatedIOBuffer(copy(io.io), copy(io.annotations))
+unsafe_write(io::AnnotatedIOBuffer, b::Ptr{UInt8}, len::UInt) = unsafe_write(io.io, b, len)
+unsafe_read(io::AnnotatedIOBuffer, b::Ptr{UInt8}, len::UInt) = unsafe_read(io.io, b, len)
+
+function write(io::AnnotatedIOBuffer, astr::Union{AnnotatedString, SubString{<:AnnotatedString}})
+    astr = AnnotatedString(astr)
+    offset = position(io.io)
+    for (region, annot) in astr.annotations
+        start, stop = first(region), last(region)
+        push!(io.annotations, (start+offset:stop+offset, annot))
+    end
+    write(io.io, String(astr))
+end
+write(io::AnnotatedIOBuffer, c::AnnotatedChar) = write(io, AnnotatedString(c))
+write(io::AnnotatedIOBuffer, x::AbstractString) = write(io.io, x)
+write(io::AnnotatedIOBuffer, s::Union{SubString{String}, String}) = write(io.io, s)
+write(io::AnnotatedIOBuffer, b::UInt8) = write(io.io, b)
+
+"""
+    read(io::AnnotatedIOBuffer, AnnotatedString)
+
+Read `io` as an `AnnotatedString`, jumping to the start if `eof(io)` holds.
```

Comment:
```suggestion
Read `io` as an `AnnotatedString`.
```

It is not valid for a method to define a function to mean something completely different from the method it overrides.

Reply:
The method no longer does this.